### PR TITLE
fix: fixing repo related publish errors

### LINKED ISSUE #7: better error message when repository field is missing
I was receiving a `400 Bad Request` because I had not set a repository field in the `package.json`.

### PR DESCRIPTION
**Pull request by soldair**
this fixes issues where no error message was printed when a repository
was missing or misconfigured. now you get something like

```
npm ERR! code E400
npm ERR! 400 Bad Request - PUT https://.... -
npm ERR!   ===============================
npm ERR!   Publish service error
npm ERR!   -------------------------------
npm ERR!   respository https://github.com/<repository> doesnt exist or <user> doesnt have access
npm ERR!   ===============================
npm ERR!

npm ERR! A complete log of this run can be found in:
npm ERR!     /..../.._34_372Z-debug.log
```
fixes #7

### CODE REVIEW

File: src/lib/write-package.ts [hunk 1]
```diff
@@ -122,7 +122,7 @@ export const writePackage = async (
       ),
       statusCode: 500,
     };
-    res.end(ret.error);
+    res.end(JSON.stringify(ret));
```
Comment (JustinBeckwith) [inline src/lib/write-package.ts]: Here and below - can we use `res.json` instead?

File: src/lib/write-package.ts [hunk 2]
```diff
@@ -150,21 +150,24 @@ export const writePackage = async (
...
     return ret;
   }
 
   let repoResp = null;
   try {
     repoResp = await github.getRepo(repo.name, user.token);
   } catch (e) {
+    console.info('failed to get repo response for '+repo.name+' '+e)
```
Comment (JustinBeckwith) [inline src/lib/write-package.ts]: Thoughts on using backticks and interpolation?

File: src/lib/write-package.ts (hunk 1 above)
Comment (soldair) [reply src/lib/write-package.ts]: I'm ok with using res.json. 

I like to imagine that using methods only available on http servers makes things easier to move around but in practice no one ever does.

File: src/lib/write-package.ts (hunk 2 above)
Comment (soldair) [reply src/lib/write-package.ts]: I'm :man_shrugging: 
if you don't have new lines in your message back ticks are nearly as easy.

**Comment by soldair** [comment]
i refactored error handling in write package to usea common function to write and format errors on the resonse. uses res.json, backticks and other suggestions. cc @JustinBeckwith

File: src/lib/write-package.ts (hunk 1 above)
Comment (JustinBeckwith) [reply src/lib/write-package.ts]: Heh, I come from the other side where I learned express far before the base level HTTP server, so I never know what's express and what's node, like the new jquery vs javascript

**Comment by bcoe** [comment]
awesome work @soldair, I like that you refactored the response logic.

**Pull request merged**